src/utils/issue.ts

```diff
@@ -107,6 +104,37 @@ export async function closePullRequestForAnIssue(context: Context, issueNumber:
   return logger.info(comment);
 }
 
+async function confirmMultiAssignment(context: Context, issueNumber: number, usernames: string[]) {
+  const { logger, payload, octokit } = context;
+
+  if (usernames.length < 2) {
+    return;
+  }
+
+  const { private: isPrivate } = payload.repository;
+
+  const {
+    data: { assignees },
+  } = await octokit.rest.issues.get({
+    owner: payload.repository.owner.login,
+    repo: payload.repository.name,
+    issue_number: issueNumber,
+  });
+
+  if (!assignees?.length) {
+    const log = logger.error("We detected that this task was not assigned to anyone. Please report this to the maintainers.", { issueNumber, usernames });
+    await addCommentToIssue(context, log?.logMessage.diff as string);
+    throw new Error(log?.logMessage.raw);
```

Comment:
consider throwing the error which `logger.error` returns and let it bubble up and add a catch at the start of the chain where it posts a comment
for reference [this](https://github.com/ubiquibot/conversation-rewards/blob/development/src/index.ts) is how it's done in `conversation-rewards`

Reply:
So in this async-await context would that just look like a try-catch at the beginning of this function and `addCommentToIssue` from the catch block?

The reference is in a workflow context, so replicating that as-is we'd wrap `runPlugin()` in `plugin.ts` and we bubble up all errors from across the logic and post from there?

Reply:
@whilefoo I have placed it in `plugin.ts` wrapping the `userStartStop()` function. The reason why is because if we do it inside `worker.ts` then we need to rebuild `context` to pass into `addCommentToIssue` otherwise split `worker.ts` so that it's globally accessible from the catch block which means two try blocks one for everything before `await request.json()` and one for everything after. 

It seems like a smoother approach handling it after `context` is built and easily accessible however I can refactor `worker.ts` if you'd prefer that way